### DISCUSSION TITLE
umlet should depend on a non-jre8 package

### DISCUSSION THREAD

**Discussion by rgl** [chocolatey-community/chocolatey-packages#1926] [Ideas]
The umlet package should be migrated away from the Oracle JRE due to licensing FUD.

I would migrate it to use the temurin8jre package; but it would be better to depend on some kind of virtual java runtime package instead of depending on a specific java distribution.

**Comment by TheCakeIsNaOH** [comment]
>  but it would be better to depend on some kind of virtual java runtime package instead of depending on a specific java distribution.

Agreed that it would be ideal, but unfortunately, Chocolatey has not implemented the `provides` element required for this type of functionality.

**Comment by rgl** [comment]
How about dropping the dependey altogether? And let the user responsible to install Java on its own?

**Reply by AdmiringWorm** [reply]
Since there are several options of Java available, I think that should be appropriate.

If removing it from the package though, there will need to be a warning outputted to the user that they need to install Java on their own.

Is this something you would be willing to look into doing?
If it is, I can approve and promote this discussion to an issue immediately

**Reply by rgl** [reply]
Yes, I will do it.

I'm not sure about the warning thou, is that something to be added into the README? In the `chocolateyInstall.ps1` as a `Write-Host` message?

**Reply by AdmiringWorm** [reply]
Not in the Readme, and not using the `Write-Host` cmdlet, but rather using the `Write-Warning` cmdlet instead.

I'll create an issue based on this discussion and reference you, please leave a comment on the issue so I can assign it to you.
